packages/ui-view/src/View/types.ts

```diff
@@ -0,0 +1,295 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import PropTypes from 'prop-types'
+import {
+  cursor as cursorPropTypes,
+  OtherHTMLAttributes
+} from '@instructure/ui-prop-types'
+import { bidirectional } from '@instructure/ui-i18n'
+import {
+  ThemeablePropTypes,
+  WithStyleProps,
+  Spacing,
+  BorderWidth,
+  BorderRadii,
+  Shadow,
+  Stacking
+} from '@instructure/emotion'
+
+export type ViewOwnProps = {
```

Comment:
Is this the correct approach? As we've seen a file that only exports types can cause issues

Reply:
I dont think it will be an issue in this case, because this file also exports a list of possible prop keys which is not a TS only thing (it is a runtime variable).